Lib/dataclasses.py

```diff
@@ -753,6 +753,8 @@ def _get_field(cls, a_name, a_type):
 def _set_new_attribute(cls, name, value):
     # Never overwrites an existing attribute.  Returns True if the
     # attribute already exists.
+    if isinstance(value, FunctionType):
+        value.__qualname__ = f"{cls.__qualname__}.{value.__name__}"
```

Comment:
Why `value.__name__` and not `name`? Are they always the same?

Reply:
For all use cases, but since it got extracted to another function which only takes `cls` and `value`, I guess there is no need to complicate that function.